ompi/mca/osc/rdma/osc_rdma_component.c

```diff
@@ -578,21 +579,21 @@ static int allocate_state_shared (ompi_osc_rdma_module_t *module, void **base, s
 
     /* CPU atomics can be used if every process is on the same node or the NIC allows mixing CPU and NIC atomics */
     module->single_node     = local_size == global_size;
-    module->use_cpu_atomics = module->single_node;
+    use_cpu_atomics = module->single_node;
 
     if (!module->single_node) {
         if (module->use_accelerated_btl) {
-            module->use_cpu_atomics = !!(module->accelerated_btl->btl_flags & MCA_BTL_ATOMIC_SUPPORTS_GLOB);
+            use_cpu_atomics = !!(module->accelerated_btl->btl_flags & MCA_BTL_ATOMIC_SUPPORTS_GLOB);
         } else {
             for (int i = 0 ; i < module->alternate_btl_count ; ++i) {
-                module->use_cpu_atomics &= !!(module->alternate_btls[i]->btl_flags & MCA_BTL_ATOMIC_SUPPORTS_GLOB);
+                use_cpu_atomics &= !!(module->alternate_btls[i]->btl_flags & MCA_BTL_ATOMIC_SUPPORTS_GLOB);
```

Comment:
There is a bug in the original implementation. I think because you are modifying this code, maybe fix it.

The bug is that for multiple nodes case, `use_cpu_atomics` has already been set to `false`. Therefore, line 589 is not-effective.  So we will need to initialize `use_cpu_atomics` to `true` before the loop.